### ISSUE TITLE
Charge Source

### ISSUE THREAD

**Issue by jmawebtech** [stripe/stripe-dotnet#1384]
Hi,

With some charges, they have a source of ACH debit. These charges must be queried using the source field. How do I query by source using the ChargeService? Here is the URL that I would use:

```c#
string url = String.Format("{0}/v1/charges?created%5Bgte%5D={1}&created%5Blte%5D={2}&limit=100&expand[]=data.invoice&expand[]=data.customer&offset={3}", m_BaseUrl, ConvertToTimestamp(beginTime), ConvertToTimestamp(endTime), i * 100);

if (!String.IsNullOrEmpty(source))
    url += "&source[object]=" + source;
```

**Comment by ob-stripe** [comment]
Hi @jmawebtech. Right now, the `ChargeListOptions` class is lacking the `Source` attribute, but I'm going to push a fix that will allow you to do something like this:

```c#
var chargeService = new ChargeService();
var listOptions = new ChargeListOptions
{
    Created = new DateRangeOptions
    {
        GreatherThanOrEqual = beginTime,
        LessThanOrEqual = endTime,
    },
};
if (!String.IsNullOrEmpty(source))
{
    listOptions.Source = new ChargeSourceListOptions
    {
        Object = source,
    };
}
listOptions.AddExpand("data.invoice");
listOptions.AddExpand("data.customer");

var charges = chargeService.List(listOptions);
```

**Comment by ob-stripe** [comment]
This is fixed as of version 21.1.0.
